utils/protocols.py

```diff
@@ -11,17 +12,35 @@ def is_10x(ontology_api: OntologyAPI, library_preparation_protocol: dict):
     library_construction = get_library_construction(library_preparation_protocol)
     if library_construction == ONTOLOGY_CITESEQ:
         return True
-    return ontology_api.is_equal_or_descendant(ONTOLOGY_10x_PARENT, library_construction)
+    return ontology_api.is_equal_or_descendant(ONTOLOGY_3PRIME_PARENT, library_construction) or \
+           ontology_api.is_equal_or_descendant(ONTOLOGY_5PRIME_PARENT, library_construction)
+
+
+def is_leaf_term(ontology_api: OntologyAPI, library_construction: str):
+    term = ontology_api.search(library_construction)
+    ontology = term['ontology_name']
+    iri = term['iri']
+    return False if ontology_api.get_descendants(ontology, iri) else True
+
+
+def version_10x_by_label(ontology_api: OntologyAPI, library_preparation_protocol: dict):
+    library_construction = get_library_construction(library_preparation_protocol)
+    term = ontology_api.search(library_construction)
+    label = term['label']
+    version = re.findall('v[23]$|$', label)[0]
+    if not version:
+        raise ProtocolError(
+            f'Could not determine library construction from library_preparation_protocol: '
+            f'{label}')
```

Comment:
is the message correct? shouldn't it say ProtocolError(f'Could not determine version from {label}') ?